### PR TITLE
Mask Online DPO objective metrics at padding

### LINKED ISSUE #6803: Online DPO: `objective/kl` and `objective/entropy` include padding tokens
### Describe the bug

In Online DPO, `objective/kl`, `objective/entropy`, `objective/non_score_reward` and `objective/rlhf_reward` sum per-token logprobs over **all** completion positions, including padding after EOS.

`trl/experimental/online_dpo/online_dpo_trainer.py:1252-1266`
```python
kl = logprobs - ref_logprobs
mean_kl = kl.sum(1).mean()
...
non_score_reward = (-self.beta * kl).sum(1)
...
mean_entropy = -logprobs.sum(1).mean()
```

`logprobs` comes from `_forward`, which returns a value at every completion position. The loss path in the same function masks these correctly:

`online_dpo_trainer.py:1217-1221`
```python
padding_mask = ~completion_mask.bool()
cr_padding_mask = padding_mask[cr_indices]
cr_logprobs_sum = (cr_logprobs * ~cr_padding_mask).sum(1)
```

but `completion_mask` is never applied on the stats path. The reported values therefore scale with how much padding a batch happens to carry, making them non-comparable across batches and across `max_new_tokens` settings. Training is unaffected — this is logging only.

Note the symptom differs from the PPO case below: Online DPO has no `INVALID_LOGPROB` sentinel, so the pad positions contribute ordinary (finite, negative) log-softmax values rather than `+1.0`. Entropy does not go impossibly negative here; it is simply inflated.

### The two sibling trainers were already fixed

- PPO: fixed in #6121 (merged 2026-07-27), which multiplies by `(~padding_mask).float()` before summing.
- XPO / Nash-MD: mask upstream with `masked_fill(padding_mask, 0.0)` (`xpo_trainer.py:298-301`) so their `objective/kl` and `objective/entropy` at `xpo_trainer.py:398-407` are already clean.

Online DPO appears to be the remaining unfixed copy of this pattern.

### Suggested fix

Apply `completion_mask` before summing, e.g.

```python
kl = (logprobs - ref_logprobs) * completion_mask
mean_kl = kl.sum(1).mean()
...
mean_entropy = -(logprobs * completion_mask).sum(1).mean()
```

Happy to open a PR aligning it with the PPO/XPO/Nash-MD copies.

### System Info

`trl` @ main (f724602), `transformers` 5.9.0, `torch` 2.12.0.

### PR DESCRIPTION
**Pull request by guptaishaan**
# What does this PR do?

Fixes #6803

Online DPO summed finite log probabilities from padded completion positions when calculating objective statistics. This applies `completion_mask` to KL and entropy inputs, which also corrects the derived non-score and RLHF rewards.

The regression test failed before the fix and passed after it. The full Online DPO test module passed on Linux with an NVIDIA A40, using CPU execution because the installed PyTorch CUDA build requires a newer driver. The PEFT, vLLM, and vision tests were skipped because their optional dependencies were not installed, and distributed training was not tested.

Thanks to @mmjerge for reporting the issue and identifying the affected metrics.

## Before submitting

- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request), Pull Request section?
- [x] Was this discussed/approved via a GitHub issue? Please add a link to it if that's the case.
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests?

No documentation update is needed for this logging fix.

## AI writing disclosure

We welcome the use of AI tools to help with contributions. For transparency and to help us improve our review process, please indicate the level of AI involvement in this PR.

- [ ] No AI usage: the PR was written entirely by a human.
- [ ] AI-assisted: some parts were suggested or improved by AI, but the PR was written and reviewed by a human.
- [x] AI-generated: the PR was mostly or fully generated by an AI tool.

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag members/contributors who may be interested in your PR.

---

> [!NOTE]
> **Low Risk**
> Small, localized change to training-step metric aggregation only; no change to the DPO loss computation path.
> 
> **Overview**
> Fixes incorrect **logged** objective statistics in `OnlineDPOTrainer` when completions are padded to a fixed length.
> 
> **KL** and **entropy** (and metrics derived from them—**non_score_reward**, **objective/rlhf_reward**) now multiply per-token log-probability terms by `completion_mask` before summing, matching how chosen/rejected log-prob sums already ignore padding. Training loss behavior is unchanged; this is a metrics/logging fix.
> 
> Adds `test_objective_metrics_ignore_padding`, which monkeypatches generation and forward passes with a known mask and asserts the expected stat values.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit e797c2c36375382d576994abd0a0ccacc8248765. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by qgallouedec** [comment]
Closing: we don't review fully AI-generated PRs from first-time contributors, see the [AI usage policy](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#ai-usage-policy).
